spring-batch-infrastructure/src/main/java/org/springframework/batch/support/transaction/TransactionAwareBufferedWriter.java

```diff
@@ -210,19 +274,22 @@ public void flush() throws IOException {
 	public void write(char[] cbuf, int off, int len) throws IOException {
 
 		if (!transactionActive()) {
-			char [] subArray = new char[len];
-			System.arraycopy(cbuf, off, subArray, 0, len);
-			byte[] bytes = new String(subArray).getBytes(encoding);
-			int length = bytes.length;
-			ByteBuffer bb = ByteBuffer.wrap(bytes);
-			int bytesWritten = channel.write(bb);
-			if(bytesWritten != length) {
-				throw new IOException("Unable to write all data.  Bytes to write: " + len + ".  Bytes written: " + bytesWritten);
+			if (len > 8192) {
```

Comment:
Same comment as previously mentioned for `8192`. And it looks like this block of code is the same as the previous one (Lines 249-260). If it is the case, can we extract this in a private method?

Reply:
I'll remove this as well.